ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -608,16 +613,26 @@ runDB standalone@DB{..} cmd =
                 (map ApplyVal bs)
                 db
     go hasFS Snap = do
-        (_, db) <- atomically $ readTVar dbState
+        -- TODO: Maybe @Flush@ should be its own command but that would require
+        -- the implementation to change so that 'ledgerDbOldest' doesn't require
+        -- a 'flush' to happen just before it is called.
+        (toFlush, bs, db') <- atomically $ do
```

Comment:
Maybe we should comment on why do we need to obtain the `toFlush`, backing store, and db state in the same atomic block.

Reply:
They probably don't need to be, since this test is single-threaded. But conceptually the read-write should be atomic.

Reply:
Maybe abstract this so that it just receives a TVar and then we factor it out here and in LgrDB?